gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapperTest.java

```diff
@@ -568,46 +569,34 @@ void When_MappingReferralRequestReferencedByReferralRequestToExternalDocumentLin
         );
         final var referralRequest = referralRequests.getFirst();
 
-        assertAll(
-            () -> assertThat(referralRequest.getSupportingInfo())
-                .hasSize(2),
-            () -> assertThat(referralRequest.getSupportingInfo().getFirst().getReferenceElement().getResourceType())
-                .isEqualTo(ResourceType.DocumentReference.name()),
-            () -> assertThat(referralRequest.getSupportingInfo().getFirst().getReferenceElement().getIdPart())
-                .isEqualTo("narrative-statement-1"),
-            () -> assertThat(referralRequest.getSupportingInfo().getLast().getReferenceElement().getResourceType())
-                .isEqualTo(ResourceType.DocumentReference.name()),
-            () -> assertThat(referralRequest.getSupportingInfo().getLast().getReferenceElement().getIdPart())
-                .isEqualTo("narrative-statement-2")
-        );
+        assertThat(referralRequest.getSupportingInfo())
+            .extracting("referenceElement.resourceType", "referenceElement.idPart")
+            .containsExactly(
+                Tuple.tuple(ResourceType.DocumentReference.name(), "narrative-statement-1"),
+                Tuple.tuple(ResourceType.DocumentReference.name(), "narrative-statement-2")
```

Comment:
Interesting feature of `assertJ`, making up for the deficiency in the Java language.

Reply:
yeah, i thought so too!